runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/work/processing/StreamingWorkScheduler.java

```diff
@@ -415,18 +415,21 @@ private ExecuteWorkResult executeWork(
 
       // Release the execution state for another thread to use.
       computationState.releaseComputationWorkExecutor(computationWorkExecutor);
+      computationWorkExecutor = null;
 
       work.setState(Work.State.COMMIT_QUEUED);
       outputBuilder.addAllPerWorkItemLatencyAttributions(work.getLatencyAttributions(sampler));
 
       return ExecuteWorkResult.create(
           outputBuilder, stateReader.getBytesRead() + localSideInputStateFetcher.getBytesRead());
     } catch (Throwable t) {
-      // If processing failed due to a thrown exception, close the executionState. Do not
-      // return/release the executionState back to computationState as that will lead to this
-      // executionState instance being reused.
-      LOG.debug("Invalidating executor after work item {} failed", workItem.getWorkToken(), t);
-      computationWorkExecutor.invalidate();
+      if (computationWorkExecutor != null) {
```

Comment:
Looks like it fixed a NPE. Was this an existing bug or exposed after this change (to MapTaskExecutor)?